### PR TITLE
Enable 4.6 cluster for odo test on pr and periodic jobs

### PR DESCRIPTION
**Pull request by prietyc123**
Enable 4.6 cluster for odo test on periodic jobs to check the stability of test according to upcoming version

Ping @amitkrout

### CODE REVIEW

**Comment by prietyc123** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/10346/rehearse-10346-periodic-ci-openshift-odo-master-v4.6-integration-e2e-periodic/1292828790027718656#1:build-log.txt%3A748 is a flake already tracked here - https://github.com/openshift/odo/issues/3610 

/retest

**Comment by amitkrout** [comment]
```
Error from server (AlreadyExists): error when creating "STDIN": subscriptions.operators.coreos.com "mongodb-enterprise" already exists
Error from server (AlreadyExists): error when creating "STDIN": subscriptions.operators.coreos.com "etcd" already exists
Error from server (AlreadyExists): error when creating "STDIN": subscriptions.operators.coreos.com "service-binding-operator" already exists
```
Because of this all the operator hub tests are failing

File: ci-operator/jobs/openshift/odo/openshift-odo-master-periodics.yaml [hunk 1]
```diff
@@ -287,3 +287,81 @@ periodics:
...
+    skip_cloning: true
+  extra_refs:
+  - base_ref: master
+    org: openshift
+    repo: odo
+  labels:
+    ci-operator.openshift.io/prowgen-controlled: "true"
+    ci-operator.openshift.io/variant: v4.6
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-odo-master-v4.6-integration-e2e-periodic
```
Comment (amitkrout) [inline ci-operator/jobs/openshift/odo/openshift-odo-master-periodics.yaml:304]: Prow job generator does not carry the notification config. First you need to add it manually then run `make jobs` to confirm the change.  
```
reporter_config:
    slack:
      channel: '#odo-notifications'
      job_states_to_report:
      - success
      - failure
      - error
      report_template: '{{if eq .Status.State "success"}} :rainbow: Job *{{.Spec.Job}}* ended with *{{.Status.State}}*. <{{.Status.URL}}|View logs> :rainbow: {{else}} :volcano: Job *{{.Spec.Job}}* ended with *{{.Status.State}}*. <{{.Status.URL}}|View logs> :volcano: {{end}}'
```

For reference check - https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/odo/openshift-odo-master-periodics.yaml#L17-L24

Comment (prietyc123) [reply ci-operator/jobs/openshift/odo/openshift-odo-master-periodics.yaml:304]: Ahh... missed it thanks for pointing it out 👍

Comment (prietyc123) [reply ci-operator/jobs/openshift/odo/openshift-odo-master-periodics.yaml:304]: done

**Comment by amitkrout** [comment]
Blocked due to https://github.com/openshift/odo/issues/3882

**Comment by amitkrout** [comment]
Config file and the generated job file along with few manual update in the job file looks good to me.

/lgtm

**Pull request merged**